### ISSUE TITLE
ListView in AnimatedCrossFade jumps to top

### ISSUE THREAD

**Issue by frogman1189** [flutter/flutter#90361]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows: ... 
3. ... 

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: CrossFadeListViewPage(),
    );
  }
}

class CrossFadeListViewPage extends StatelessWidget {
  const CrossFadeListViewPage({Key? key}) : super(key: key);

  static const routeName = 'CrossFadeListViewPage';

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: CrossFadeListViewBody(),
    );
  }
}

class CrossFadeListViewBody extends StatefulWidget {
  const CrossFadeListViewBody({Key? key}) : super(key: key);

  @override
  State<CrossFadeListViewBody> createState() => _CrossFadeListViewBodyState();
}

class _CrossFadeListViewBodyState extends State<CrossFadeListViewBody> {
  bool change = true;

  @override
  Widget build(BuildContext context) {
    return Center(
      child: SizedBox(
        height: 300,
        child: GestureDetector(
          onTap: () => setState(() => change = !change),
          child: AnimatedCrossFade(
            duration: Duration(seconds: 1),
            crossFadeState: change ? CrossFadeState.showFirst : CrossFadeState.showSecond,
            firstChild: Container(height: 100, width: 100, color: Colors.green),
            secondChild: ListView.builder(
              shrinkWrap: true,
              itemBuilder: (context, index) => Container(
                height: 50,
                color: index % 2 == 0 ? Colors.blue : Colors.blueAccent,
                child: Center(child: Text('child #$index')),
              ),
              itemCount: 20,
            ),
          ),
        ),
      ),
    );
  }
}
```
- click green box
- scroll to the bottom of the list
- click the list

**Expected results:** 
list stays in same place and crossfades to green box

**Actual results:** 
list jumps to top and crossfades to green box

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, 2.5.0, on macOS 11.5.2 20G95 darwin-arm, locale en-NZ)
    • Flutter version 2.5.0 at /Users/user/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (12 days ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 52A31433-1540-434B-A44C-3F4E46F13691 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `master` and `stable` with sample provided by author. 

<details>
<summary>video reproducing issue</summary>

https://user-images.githubusercontent.com/88313112/133991296-389ef070-3c37-4e8e-bf38-255df40c973a.mp4

</details>

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel master, 2.6.0-6.0.pre.147, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-6.0.pre.147 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 83dfb2237a (2 days ago), 2021-09-17 23:48:04 -0400
    • Engine revision 31792e0340
    • Dart version 2.15.0 (build 2.15.0-120.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • M2007J20CG (mobile) • 5dd3be00 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)     • macos    • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)        • chrome   • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

```console
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.5.1 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (3 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • M2007J20CG (mobile) • 5dd3be00 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)     • macos    • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)        • chrome   • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```
</details>

**Comment by nt4f04uNd** [comment]
Duplicate of https://github.com/flutter/flutter/issues/82614

**Comment by Piinks** [comment]
@frogman1189 can you look and see if #82614 is the same issue? If so, we can close this and follow up there. :)

**Comment by frogman1189** [comment]
Hi @Piinks, issue #82614 is definitely occurring in my example, however it's not what I was originally trying to show and It's hard for me to tell whether the list view placement is also caused by it. I'd be fine with closing this for now and then double checking it after #82614 has been fixed to make sure it also fixed this

**Comment by frogman1189** [comment]
Closing in favour of #82614
